activation/nipost.go

```diff
@@ -317,7 +318,10 @@ func (nb *NIPostBuilder) submitPoetChallenge(
 	if err != nil {
 		return nil, &PoetSvcUnstableError{msg: "failed to get PoET service ID", source: err}
 	}
-	logger := nb.log.WithContext(ctx).WithFields(log.String("poet_id", hex.EncodeToString(poetServiceID.ServiceID)))
+	logger := nb.log.With(
+		log.ZContext(ctx),
+		zap.String("poet_id", hex.EncodeToString(poetServiceID.ServiceID)),
+	)
```

Comment:
How about logging the URL instead?
- more human-friendly - easier to distinguish poets in logs
- avoids eager encoding in hex

```suggestion
	logger := nb.log.With(log.ZContext(ctx), zap.String("poet", client.Address()))
```